pkg/config/validation/validation.go

```diff
@@ -1953,12 +1955,12 @@ func validatePolicyTargetReference(selector *type_beta.WorkloadSelector, targetR
 		v = appendErrorf(v, "policyTargetReference.namespace must be set")
 	}
 	// Currently, gateway.networking.k8s.io is the only valid Group for this field.
-	if targetRef.Group != defaultGatewayGroup {
-		v = appendErrorf(v, "policyTargetReference.group is incorrect; expected: %v, got: %v", defaultGatewayGroup, targetRef.Group)
+	if targetRef.Group != defaultGatewayGVK.Group {
```

Comment:
nit: probably best to just do an OR and return a single error about both group and kind not matching

Reply:
I can update that. But how exactly would that error look? I want it to be descriptive enough that it informs the user exactly what they can do to fix it. Were you thinking of something along the lines of this?:

```go
        targetRefGVK := config.GroupVersionKind{
		Group: targetRef.group,
		Kind:  targetRef.kind,
	}
	if targetRef.Group != defaultGatewayGVK.Group || targetRef.Kind != defaultGatewayGVK.Kind {
		v = appendErrorf(v, "policyTargetReference Group and Kind don't match; expected: %v, got: %v", defaultGatewayGVK, targetRefGVK )
	}
```

Reply:
Yeah exactly like that